### DISCUSSION TITLE
Is there a way to access ALL members of a family?

### DISCUSSION THREAD

**Discussion by mark8044** [rrousselGit/riverpod#1099] [Q&A] [answered]
Right now Im using a family of providers as a way to compartmentalize my data of similar type into different easily accessible areas.  Its working like an absolute charm.  I can access pieces that I need in a very database like way.

However, is there any way to access all family members at the same time, like a wildcard?

My provider is setup as follows:
```
final newsProvider = ChangeNotifierProvider.family<NewsData, String>((ref, id) {
  return NewsData();
});
```

And I access each one, or make one do whatever I want as follows:
```
ref.read(newsProvider('News1')).getMyCustomDataFromHttp;
```

```
ref.read(newsProvider('News2')).getMyCustomDataFromHttp;
```

In this case, doing .` getMyCustomDataFromHttp`, is a void function that fetches remote Http data and places it into a `myData` variable which is then accessed via a getter `ref.read(newsProvider('News1')).myData`;

Now a user may have potentially 100s of `NewsX` providers outlined like above.  

Would there be any way to wildcard `.read` on all of them to force them all to do a `getMyCustomDataFromHttp` and fetch their respective data from Http?

For example, 

```
ref.read(newsProvider.all).getMyCustomDataFromHttp;
```

**Comment by rrousselGit** [comment, marked as answer]
There is no such thing, no.

The solution is to make a separate provider that obtains all the ids from somewhere, and combine all the values

**Reply by mark8044** [reply]
Ok thanks

**Reply by shtse8** [reply]
> There is no such thing, no.
> 
> The solution is to make a separate provider that obtains all the ids from somewhere, and combine all the values

Is there a reason why this approach is not implemented? For some server use cases (as Riverpod allows usage in Dart), it's useful to manage member disposals manually using keepAlive(). For example, when the number of members exceeds a certain threshold, you might want to dispose of the earliest one. However, since we cannot directly access the members, we need to manually manage the member list to achieve this goal.

**Reply by a1573595** [reply]
It would be great if this feature could be implemented.
